lib/any.js

```diff
@@ -683,6 +683,21 @@ module.exports = internals.Any = class {
             }
         }
 
+        if (this._settings) {
+
+            const optionSet = ['abortEarly', 'convert', 'allowUnknown', 'skipFunctions', 'stripUnknown', 'language', 'presence', 'context', 'noDefaults'];
+            if (optionSet.some((option) => this._settings.hasOwnProperty(option))) {
```

Comment:
Yes, you're right. This was the cleaner option I could come up with, since `this._settings` also contains `ref` info from the `.when` method. 

The first `hasOwnProperty` check in the conditional will only fire off for the first `option` found (if any), and you are right - that was to prevent setting the empty object.

The second `hasOwnProperty` is to ensure that only the options specified [here](https://github.com/hapijs/joi/blob/master/API.md#validatevalue-schema-options-callback) can get reported by `describe`.